Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/TapToPlace.cs

```diff
@@ -0,0 +1,373 @@
+                // The object's layer is changed to IgnoreRaycast in this method, which means the game object cannot receive focus.
+                // Without a global handler, the game object would not receive pointer events.
+                CoreServices.InputSystem?.RegisterHandler<IMixedRealityPointerHandler>(this);
+            }
+        }
+
+        /// <summary>
+        /// Stop the placement of a game object without the need of the OnPointerClicked event. 
+        /// </summary>
+        public void StopPlacement()
+        {
+            // Added for code configurability to avoid multiple calls to StopPlacement in a row
+            if (IsBeingPlaced)
+            {
+                // Change the game object layer back to the game object's layer on start
+                gameObject.layer = GameObjectLayer;
+
+                SolverHandler.UpdateSolvers = false;
+
+                IsBeingPlaced = false;
+
+                OnPlacingStopped?.Invoke();
+
+                CoreServices.InputSystem?.UnregisterHandler<IMixedRealityPointerHandler>(this);
+            }
+        }
+
+        /// <inheritdoc/>
+        public override void SolverUpdate()
+        {
+            // Make sure the Transform target is not null, added for the case where auto start is true 
+            // and the tracked target type is the controller ray, if the hand is not in the frame we cannot
+            // calculate the position of the object
+            if (SolverHandler.TransformTarget != null)
+            {
+                PerformRaycast();
+                SetPosition();
+                SetRotation();
+            }
+        }
+
+        protected virtual void PerformRaycast()
+        {
+            // The transform target is the transform of the TrackedTargetType, i.e. Controller Ray, Head or Hand Joint
+            var transform = SolverHandler.TransformTarget;
+
+            Vector3 origin = transform.position;
+            Vector3 endpoint = transform.position + transform.forward;
+            CurrentRay.UpdateRayStep(ref origin, ref endpoint);
+
+            // Check if the current ray hits a magnetic surface
+            DidHitSurface = MixedRealityRaycaster.RaycastSimplePhysicsStep(CurrentRay, MaxRaycastDistance, MagneticSurfaces, false, out CurrentHit);  
+        }
+
+        /// <summary>
+        /// Change the position of the game object if there was a hit, if not then place the object at the default distance
+        /// relative to the TrackedTargetType origin position
+        /// </summary>
+        protected virtual void SetPosition()
+        {
+            if (DidHitSurface)
+            {
+                // Take the current hit point and add an offset relative to the surface to avoid half of the object in the surface
+                GoalPosition = CurrentHit.point;  
+                AddOffset(CurrentHit.normal * SurfaceNormalOffset);
+
+#if UNITY_EDITOR
```

Comment:
Do we still want to keep the debug code?

Reply:
It was useful for debugging but not needed.  It is only useful if there is a hit on a surface and `RotateAccordingToSurface` is true. I liked the visual feedback of drawing the normal of the surface hit on non-uniform surfaces.

Reply:
Okay, up to you which way to go here, I'm fine either way.

Reply:
I am going to keep it.